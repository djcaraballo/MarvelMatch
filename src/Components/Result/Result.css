@import url('https://fonts.googleapis.com/css?family=Fjalla+One|Orbitron');

.match-heading {
  margin-top: 0;
}

.result {
  text-align: center;
  margin: 1em auto;
  height: 120%;
  background: #7d0000;
}

.match-name {
  font-size: 2.5em;
  font-family: 'Orbitron', sans-serif;
  color: #fff;
}

.char-description {
  font-family: 'Orbitron', sans-serif;
  font-size: 1em;
  line-height: 1.5em;
  color: #fff;
  width: 75%;
  margin: 1em auto;
}

.bar-graph {
  list-style: none;
  padding: 20px 0 0 0;
  width: 27%;
  margin: auto;
  text-align: center;
}

ul.bar-graph li {
  height: 2.5rem;
  color: #fff;
  font-style: italic;
  font-weight: bolder;
  font-size: 1.5em;
  line-height: 1.5em;
  padding: 0px 20px;
  margin: 5px auto;
}

/*.red {
  background: red;
  width: 69%;
}

.orange {
  background: orange;
  width: 19%;
}

.yellow {
  background: yellow;
  width: 38%;
}

.green {
  background: green;
  width: 55%;
}

.blue {
  background: blue;
  width: 60%;
}

.violet {
  background: purple;
  width: 100%;
}*/