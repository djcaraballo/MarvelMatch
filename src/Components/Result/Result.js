import React from 'react'
import PropTypes from 'prop-types'
import './Result.css'

const Result = ({ quizResult, characterMatch }) => {
  return (
    <div className='result'>
      <h1 className="match-heading">You match with...</h1>
      <h2 className="match-name">{quizResult}</h2>
      <img src={characterMatch.imageURL} alt={characterMatch.name}/>
      <p className="char-description">{characterMatch.description}</p>
      <ul className="bar-graph">
        <li className="red">Intelligence: {characterMatch.powerstats.intelligence}</li>
        <li className="orange">Strength: {characterMatch.powerstats.strength}</li>
        <li className="yellow">Speed: {characterMatch.powerstats.speed}</li>
        <li className="green">Durability: {characterMatch.powerstats.durability}</li>
        <li className="blue">Power: {characterMatch.powerstats.power}</li>
        <li className="violet">Combat: {characterMatch.powerstats.combat}</li>
      </ul>
    </div>
  )
}

Result.propTypes = {
  quizResult: PropTypes.string
}

export default Result
