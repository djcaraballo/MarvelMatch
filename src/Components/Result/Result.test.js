import React from 'react'
import Result from './Result'
import { shallow } from 'enzyme'

describe('Result Component', () =>{
  let wrapper
  let mockQuizResult = ''

  beforeEach(() => {
    wrapper = shallow(<Result quizResult={mockQuizResult} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})