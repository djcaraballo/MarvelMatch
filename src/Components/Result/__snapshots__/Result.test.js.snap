// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component should match the snapshot 1`] = `undefined`;
